src/ssl.c

```diff
@@ -30149,7 +30149,7 @@ static int set_curves_list(WOLFSSL* ssl, WOLFSSL_CTX *ctx, const char* names)
     char name[MAX_CURVE_NAME_SZ];
     byte groups_len = 0;
 #ifdef WOLFSSL_SMALL_STACK
-    void *heap = ssl? ssl->heap : ctx ? ctx->heap : NULL; (void)heap;
+    void *heap = ssl? ssl->heap : ctx ? ctx->heap : NULL;
```

Comment:
This was added specifically for (https://github.com/wolfSSL/wolfssl/pull/6995). Would like @gojimmypi to comment before undoing the fix.

Reply:
Thanks for asking. Good catch on a recent change.

> remove frivolous (void)heap

I realize that it may _look_ frivolous, however none of the usual explicit syntax adjustments would appease the ESP-IDF v5.1 Espressif compiler. I should have put a comment there as such. 

If revisiting this particular line, I've been trying not to assign variables with an expression on the same line as declaring them, per a suggestion from @dgarske.  I added the `(void)heap` as it was least intrusive.

Reply:
Thanks for connecting those dots.  Reviewing `types.h` I've already found and fixed several macro implementations of `XMALLOC()` that were missing the needed `(void)`ings.  Once I'm done I think we'll have it covered.

Reply:
Yeah this is the way to do it.  And for the record, @julek-wolfssl originated the code pattern for us, and I just expanded its coverage.  Works a treat.